x-pack/plugins/apm/public/components/app/error_group_overview/index.tsx

```diff
@@ -15,20 +15,25 @@ import {
 import { i18n } from '@kbn/i18n';
 import React from 'react';
 import { useTrackPageview } from '../../../../../observability/public';
+import { useApmServiceContext } from '../../../context/apm_service/use_apm_service_context';
 import { useUrlParams } from '../../../context/url_params_context/use_url_params';
+import { useApmParams } from '../../../hooks/use_apm_params';
 import { useErrorGroupDistributionFetcher } from '../../../hooks/use_error_group_distribution_fetcher';
 import { useFetcher } from '../../../hooks/use_fetcher';
 import { ErrorDistribution } from '../error_group_details/Distribution';
 import { ErrorGroupList } from './List';
 
-interface ErrorGroupOverviewProps {
-  serviceName: string;
-}
+export function ErrorGroupOverview() {
+  const { serviceName } = useApmServiceContext();
+
+  const {
+    query: { environment, kuery, sortField, sortDirection },
+  } = useApmParams('/services/:serviceName/errors');
 
-export function ErrorGroupOverview({ serviceName }: ErrorGroupOverviewProps) {
   const {
-    urlParams: { environment, kuery, start, end, sortField, sortDirection },
+    urlParams: { start, end },
```

Comment:
Are we still relying on `urlParams` magic for automatically getting `start`/`end`? Should we have some kind of helper like `const { start, end } = getTimeRange({ rangeFrom, rangeTo });` to make it more explicit or maybe do the same thing with `useApmParams` that we do with `useUrlParams`?

Reply:
Yeah, we should probably have a useTimeRange or a getTimeRange function. After this is in I'll create some follow-up issues.